plugins/ti-tps6598x/fu-ti-tps6598x-pd-device.c

```diff
@@ -0,0 +1,196 @@
+ *
+ * SPDX-License-Identifier: LGPL-2.1+ OR MIT
+ */
+
+#include "config.h"
+
+#include "fu-ti-tps6598x-common.h"
+#include "fu-ti-tps6598x-device.h"
+#include "fu-ti-tps6598x-firmware.h"
+#include "fu-ti-tps6598x-pd-device.h"
+
+struct _FuTiTps6598xPdDevice {
+	FuDevice parent_instance;
+	guint8 target;
+};
+
+G_DEFINE_TYPE(FuTiTps6598xPdDevice, fu_ti_tps6598x_pd_device, FU_TYPE_DEVICE)
+
+// FIXME move to common code
+static gchar *
+fu_byte_array_to_string(GByteArray *buf)
+{
+	g_autoptr(GString) str = g_string_new(NULL);
+	g_return_val_if_fail(buf != NULL, NULL);
+	for (guint i = 0; i < buf->len; i++)
+		g_string_append_printf(str, "%02x", buf->data[i]);
+	return g_string_free(g_steal_pointer(&str), FALSE);
+}
+
+static gboolean
+fu_ti_tps6598x_pd_device_probe(FuDevice *device, GError **error)
+{
+	FuTiTps6598xPdDevice *self = FU_TI_TPS6598X_PD_DEVICE(device);
+	FuTiTps6598xDevice *parent = FU_TI_TPS6598X_DEVICE(fu_device_get_parent(device));
+	g_autofree gchar *name = g_strdup_printf("TPS6598X PD#%u", self->target);
+	g_autofree gchar *logical_id = g_strdup_printf("PD%u", self->target);
+
+	/* do as few register reads as possible as they are s...l...o...w... */
+	fu_device_set_name(device, name);
+	fu_device_set_logical_id(device, logical_id);
+
+	/* fake GUID */
+	fu_device_add_instance_u16(device, "VID", fu_usb_device_get_vid(FU_USB_DEVICE(parent)));
+	fu_device_add_instance_u16(device, "PID", fu_usb_device_get_vid(FU_USB_DEVICE(parent)));
+	fu_device_add_instance_u8(device, "PD", self->target);
+	return fu_device_build_instance_id(device, error, "USB", "VID", "PID", "PD", NULL);
+}
+
+static gboolean
+fu_ti_tps6598x_pd_device_setup(FuDevice *device, GError **error)
+{
+	FuTiTps6598xPdDevice *self = FU_TI_TPS6598X_PD_DEVICE(device);
+	FuTiTps6598xDevice *parent = FU_TI_TPS6598X_DEVICE(fu_device_get_parent(device));
+	g_autoptr(GByteArray) buf = NULL;
+	g_autofree gchar *version = NULL;
+	g_autofree gchar *config = NULL;
+
+	/* register reads are s...l...o...w... */
+	buf = fu_ti_tps6598x_device_read_target_register(parent,
+							 self->target,
+							 TI_TPS6598X_REGISTER_VERSION,
+							 4,
+							 error);
+	if (buf == NULL)
+		return FALSE;
+	version = g_strdup_printf("%02X%02X.%02X.%02X",
+				  buf->data[3],
+				  buf->data[2],
+				  buf->data[1],
+				  buf->data[0]);
+	fu_device_set_version_format(device, FWUPD_VERSION_FORMAT_PLAIN);
+	fu_device_set_version(device, version);
+
+	/* the PD OTP config should be unique enough */
+	buf = fu_ti_tps6598x_device_read_target_register(parent,
+							 self->target,
+							 TI_TPS6598X_REGISTER_OTP_CONFIG,
+							 12,
+							 error);
+	if (buf == NULL)
+		return FALSE;
+	config = fu_byte_array_to_string(buf);
+	fu_device_add_instance_strup(device, "CONFIG", config);
+
+	/* success */
+	return fu_device_build_instance_id(device, error, "USB", "VID", "PID", "CONFIG", NULL);
+}
+
+static gboolean
+fu_ti_tps6598x_pd_device_byte_array_is_interesting(GByteArray *buf)
```

Comment:
Is it really a good name for this function? ;)

Reply:
`fu_ti_tps6598x_pd_device_byte_array_is_zero` perhaps?

Reply:
Sounds good.

Reply:
Fixed.